### PR TITLE
[MM-1081]: Fixed the rendering of hyperlink in tooltip

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
### Summary
- Fixed the rendering of hyperlink in tooltip
-  Also fixes the rendering of markdowns in issue events related webhook notifications
#### Existing
Screenshot from 2024-09-19 23-38-46
Screenshot from 2024-09-19 16-37-55
#### Updated
Screenshot from 2024-09-19 23-35-35
Screenshot from 2024-09-19 16-36-00

### CODE REVIEW

**Comment by raghavaggarwal2308** [comment]
@Kshitij-Katiyar THe screenshots are not of tooltips

**Comment by Kshitij-Katiyar** [comment]
> @Kshitij-Katiyar THe screenshots are not of tooltips

Fixed

File: webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx [hunk 1]
```diff
@@ -211,7 +228,7 @@ export default class TicketPopover extends React.PureComponent<Props, State> {
                         {this.tagTicketStatus(ticketDetails.statusKey)}
                     </div>
                     <div className='popover-body__description'>
-                        {ticketDetails.description && `${ticketDetails.description.substring(0, maxTicketDescriptionLength).trim()}${ticketDetails.description.length > maxTicketDescriptionLength ? '...' : ''}`}
+                        <div dangerouslySetInnerHTML={{__html: preProcessDescription(ticketDetails.description)}}/>
```
Comment (raghavaggarwal2308) [inline webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx]: What is the need or using dangerouslySetInnerHTML attribute?

File: webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx [hunk 2]
```diff
@@ -35,6 +35,23 @@ const myStatusClasses: Record<string, string> = {
     [myStatus.DONE]: 'ticket-status--done',
 };
 
+function preProcessDescription(description: string): string {
```
Comment (raghavaggarwal2308) [inline webapp/src/components/jira_ticket_tooltip/jira_ticket_tooltip.tsx]: Cant we do it on backend side? As the data is coming from there only. We already have a function there to handle all this. Also we are supporting just links here, we need to handle other cases as well

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

**Review by raghavaggarwal2308 — APPROVED** [review]

**Comment by Kshitij-Katiyar** [comment]
Created a PR on mattermost remote with all these changes hence closing this one

**Pull request closed without merging**